server/graphql/v2/mutation/RootMutations.ts

```diff
@@ -49,6 +50,39 @@ const BanAccountResponse = new GraphQLObjectType({
  * Root mutations
  */
 export default {
+  editAccountFlags: {
+    type: new GraphQLNonNull(Account),
+    description: '[Root only] Edits account flags (deleted, banned, archived, trusted host)',
+    args: {
+      account: {
+        type: new GraphQLNonNull(AccountReferenceInput),
+        description: 'Account to change the flags for',
+      },
+      isArchived: {
+        type: GraphQLBoolean,
+        description: 'Specify whether the account is archived',
+      },
+      isTrustedHost: {
+        type: GraphQLBoolean,
+        description: 'Specify whether the account is a trusted host',
+      },
+    },
+    async resolve(_: void, args, req: express.Request): Promise<Record<string, unknown>> {
+      checkRemoteUserCanRoot(req);
+      const account = await fetchAccountWithReference(args.account, { throwIfMissing: true, paranoid: false });
+
+      if (args.isArchived && !account.deactivatedAt) {
+        await archiveCollective(_, account, req);
+      } else if (args.isArchived === false && account.deactivatedAt) {
+        await unarchiveCollective(_, account, req);
+      }
+
+      if (Boolean(args.isTrustedHost) !== Boolean(account.data?.isTrustedHost)) {
+        await account.update({ data: { ...account.data, isTrustedHost: args.isTrustedHost } });
+      }
```

Comment:
Same issue as https://github.com/opencollective/opencollective-api/pull/7981#discussion_r980865197, we don't want hosts to be marked as "Untrusted" if I call the mutation like this:

```es6
editAccountFlags(account: 'foundation', isArchived: false)
```

So:
```suggestion
      if (!isNil(args.isTrustedHost) && Boolean(args.isTrustedHost) !== Boolean(account.data?.isTrustedHost)) {
        await account.update({ data: { ...account.data, isTrustedHost: args.isTrustedHost } });
      }
```